CHANGELOG.md

```diff
@@ -1,3 +1,8 @@
+## 2.0.4 (WIP)
```

Comment:
We should probably try to add a scenario test for this.

Reply:
to validate this fix we need to create a simulation where files are being updated in parallel through other means on container. Any of the test infra that we have does not support such case of now. I am thinking over how to add case for this, but as of now no clue on this.